subsys/usb/class/usb_dfu.c

```diff
@@ -388,7 +388,8 @@ static int dfu_class_handle_req(struct usb_setup_packet *pSetup,
 		case dfuIDLE:
 			SYS_LOG_DBG("DFU_DNLOAD start");
 			dfu_reset_counters();
-			if (dfu_data.flash_addr != FLASH_AREA_IMAGE_1_OFFSET) {
```

Comment:
I guess L:398 `boot_erase_img_bank(FLASH_AREA_IMAGE_1_OFFSET)`
should also be fixed. @zhaozhongchen Can you please add CONFIG_FLASH_BASE_ADDRESS there and test it?

Reply:
actually hello_sigend.bin can download but after reset the console print some characters that can't recognized. i will check if BASE_ADDRESS can be worked. is the hello_signed.bin always downloaded to slot 1, i thought i should check the flash's contains.